Run test_modern.py by itself with tox

I wanted to take a stab at this roadmap item:

> 3. Add additional CLI options to delete debris from builds, testing and packaging (build/, .cache/, dist/, .pytest_cache/, .tox/ and free-form targets).

I modified some of the tests in test_modern.py and attempted to run just the tests in that module like so:

```
$ tox -e py39 -- tests/test_modern.py
```

This ran all the tests, so I added `{posargs}` to the pytest command in tox.ini:

```
commands = pytest {posargs}
```

Now, running the same command produces an error:

```
$ tox -e py39 -- tests/test_modern.py
GLOB sdist-make: /Users/dave.kennedy/Code/pyclean/setup.py
py39 inst-nodeps: /Users/dave.kennedy/Code/pyclean/.tox/.tmp/package/1/pyclean-2.0.0.zip
py39 installed: attrs==20.3.0,cli-test-helpers==1.0.1,iniconfig==1.1.1,packaging==20.9,pluggy==0.13.1,py==1.10.0,pyclean @ file:///Users/dave.kennedy/Code/pyclean/.tox/.tmp/package/1/pyclean-2.0.0.zip,pyparsing==2.4.7,pytest==6.2.2,toml==0.10.2
py39 run-test-pre: PYTHONHASHSEED='388407112'
py39 run-test: commands[0] | pytest tests/test_modern.py
================================================= test session starts ==================================================
platform darwin -- Python 3.9.1, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /Users/dave.kennedy/Code/pyclean/.tox/py39/bin/python
cachedir: .tox/py39/.pytest_cache
rootdir: /Users/dave.kennedy/Code/pyclean, configfile: tox.ini
collected 0 items / 1 error                                                                                            

======================================================== ERRORS ========================================================
________________________________________ ERROR collecting tests/test_modern.py _________________________________________
tests/test_modern.py:50: in <module>
    ???
E   AttributeError: module 'pyclean' has no attribute 'modern'
```

The offending line is here:

```
@patch.object(pyclean.modern.logging, 'basicConfig')
```

I tried the following commands as well:

* tox -e py39 -- ./tests/test_modern.py
* tox -e py39 -- pyclean/tests/test_modern.py
* tox -e py39 -- test_modern.py

I also tried various combinations of the changedir and basetemp and config values in tox.ini, e.g.:

```
[testenv]
description = Unit tests
deps =
    cli-test-helpers
    py27: mock
    pypy2: mock
    pytest
changedir = tests
commands = pytest --basetemp="{envtmpdir}" {posargs}
```

What am I doing wrong here? It's obviously not a big deal but would be nice to run only the tests I'm changing.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -15,7 +15,7 @@ jobs:
       fail-fast: false
       matrix:
         platform:
-        - ubuntu-latest
+        - ubuntu-18.04
         - macos-latest
         - windows-latest
         python-version:

--- a/tests/test_modern.py
+++ b/tests/test_modern.py
@@ -13,7 +13,8 @@ except ImportError:  # Python 2.7, PyPy2
 
 from cli_test_helpers import ArgvContext
 
-import pyclean
+import pyclean.cli
+import pyclean.modern
 
 
 @pytest.mark.skipif(sys.version_info < (3,), reason="requires Python 3")

--- a/tox.ini
+++ b/tox.ini
@@ -17,7 +17,7 @@ deps =
     py27: mock
     pypy2: mock
     pytest
-commands = pytest
+commands = pytest {posargs}
 
 [testenv:bandit]
 description = PyCQA security linter
@@ -83,5 +83,4 @@ output-format = colorized
 [pytest]
 addopts =
     --color=yes
-    --strict
     --verbose

--- a/tox.ini
+++ b/tox.ini
@@ -21,15 +21,15 @@ commands = pytest {posargs}
 
 [testenv:bandit]
 description = PyCQA security linter
-deps = bandit<1.6.0
-commands = bandit -r . --ini tox.ini
+deps = bandit
+commands = bandit --ini tox.ini {posargs:-r pyclean setup}
 
 [testenv:clean]
 description = Clean up bytecode and build artifacts
 deps = pyclean
 commands =
     pyclean {toxinidir}
-    rm -rf .tox/ build/ dist/ pyclean.egg-info/
+    rm -rf .tox/ build/ dist/ pyclean.egg-info/ .pytest_cache/ pytestdebug.log
 whitelist_externals =
     rm
 

--- a/tox.ini
+++ b/tox.ini
@@ -43,6 +43,11 @@ description = Check for errors and code smells
 deps = pylint
 commands = pylint --rcfile tox.ini {posargs:pyclean setup}
 
+[testenv:license]
+description = Manage license compliance
+deps = reuse
+commands = reuse {posargs:lint}
+
 [testenv:readme]
 description = Ensure README renders on PyPI
 deps = twine

